metrics/ci_integrations/lib/source/buildkite/config/validation_delegate/buildkite_source_validation_delegate.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:ci_integration/client/buildkite/buildkite_client.dart';
+import 'package:ci_integration/client/buildkite/models/buildkite_organization.dart';
+import 'package:ci_integration/client/buildkite/models/buildkite_pipeline.dart';
+import 'package:ci_integration/client/buildkite/models/buildkite_token.dart';
+import 'package:ci_integration/client/buildkite/models/buildkite_token_scope.dart';
+import 'package:ci_integration/integration/interface/source/config/validation_delegate/source_validation_delegate.dart';
+import 'package:ci_integration/source/buildkite/strings/buildkite_strings.dart';
+import 'package:ci_integration/util/authorization/authorization.dart';
+import 'package:ci_integration/util/model/interaction_result.dart';
+
+/// A [SourceValidationDelegate] for the Buildkite source integration.
+class BuildkiteSourceValidationDelegate implements SourceValidationDelegate {
+  /// A [List] containing all required [BuildkiteTokenScope]s.
+  static const List<BuildkiteTokenScope> _requiredTokenScopes = [
+    BuildkiteTokenScope.readBuilds,
+  ];
+
+  /// A [BuildkiteClient] used to perform calls to the Buildkite API.
+  final BuildkiteClient _client;
+
+  /// Creates an instance of the [BuildkiteSourceValidationDelegate]
+  ///
+  /// Throws an [ArgumentError] if the given [_client] is `null`.
+  BuildkiteSourceValidationDelegate(this._client) {
+    ArgumentError.checkNotNull(_client);
+  }
+
+  @override
+  Future<InteractionResult<BuildkiteToken>> validateAuth(
+    AuthorizationBase auth,
+  ) async {
+    final tokenInteraction = await _client.fetchToken(auth);
+
+    if (tokenInteraction.isError || tokenInteraction.result == null) {
+      return const InteractionResult.error(
+        message: BuildkiteStrings.tokenInvalid,
+      );
+    }
+
+    final token = tokenInteraction.result;
+    final tokenScopes = token.scopes ?? [];
+
+    final containsRequiredScopes = _requiredTokenScopes.every(
+      (element) => tokenScopes.contains(element),
+    );
```

Comment:
Would it be great to check somewhere here that the token has the `read-artifacts` permission? I believe if the given token does not have such permission - we should return interaction with the message containing a warning about the user should run the `sync` command with the `no-coverage` flag. What do you think?